app/js/lib/foxbox.js

```diff
@@ -537,4 +537,31 @@ export default class Foxbox extends Service {
   setTag() {
     return db.setTag.apply(db, arguments);
   }
+
+  subscribeForNotifications(resubscribe = false) {
+    if (!navigator.serviceWorker) {
+      return Promise.reject('No service worker supported');
+    }
+
+    if (settings.pushEndpoint && settings.pushPubKey && !resubscribe) {
+      return Promise.resolve();
+    }
+
+    return navigator.serviceWorker.ready.then((reg) => {
+      reg.pushManager.subscribe({userVisibleOnly: true}).then((subscription) => {
```

Comment:
nit: can you insert a line break before `.then(...`? See the rest of the file for coding style.